backend/local/metadata.go

```diff
@@ -72,12 +72,12 @@ func (o *Object) parseMetadataInt(m fs.Metadata, key string, base int) (result i
 	value, ok := m[key]
 	if ok {
 		var err error
-		result64, err := strconv.ParseInt(value, base, 64)
+		parsed, err := strconv.ParseInt(value, base, 32)
```

Comment:
I don't think this is the right fix as `int` is 64 bit on most platforms but 32 on some.

I think the right fix is probably using `strconv.Atoi` which hopefully deals with the sizes of integers properly. Looking at [the source of Atoi](https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/strconv/atoi.go;l=242) I think it does the right thing.

Reply:
Right, I was too quick on this one, and thinking C++ types.

I can use `strconv.ParseInt` as well, supplying bit size `0` does the same thing - it defaults to `strconv.IntSize` and effectively returns an `int` value within the `int64` return type. It is documented that "Atoi is equivalent to ParseInt(s, 10, 0), converted to type int", however it is faster, it has  a "fast path" and then a slow path where it do call `ParseInt` internally. In this specific case, though, our function `parseMetadataInt` takes `base` as a parameter, so sending that directly into `parseInt` fits best.